packetizer.go

```diff
@@ -1,64 +1,89 @@
 package rpc
 
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	"github.com/ugorji/go/codec"
+)
+
 type packetizer interface {
-	Packetize() error
+	NextFrame() (RPCMessage, error)
 }
 
 type packetHandler struct {
-	receiver receiver
-	dec      byteReadingDecoder
+	dec           decoder
+	reader        io.Reader
+	frameDecoders decoderPool
+	protocols     *protocolHandler
+	calls         *callContainer
+	mtx           sync.Mutex
```

Comment:
I think I added it speculatively before finishing the refactor. I'll go through to make sure it's safe to remove.